### PR TITLE
feature: enable extractCSS in dev

### LINKED ISSUE #2423: css extracted to the css file
[I look forward to css extracted to the css file](https://github.com/nuxt/nuxt.js/issues/1533), because of such a use, not recognized in China Baidu seo, or look forward to a method to achieve css extraction, do not need to be embedded in the page css

image
image
image

### PR DESCRIPTION
**Pull request by clarkdo**
#### Resolve #2423

### CODE REVIEW

File: lib/builder/webpack/style-loader.js [hunk 1]
```diff
@@ -45,15 +45,16 @@ module.exports = function styleLoader(ext, loaders = [], isVueLoader = false) {
...
+  if (this.options.build.extractCSS) {
+    const extraction = ExtractTextPlugin.extract({
       fallback: vueStyleLoader,
       use: [
         cssLoader,
         postcssLoader,
         ...loaders
       ].filter(l => l)
     })
+    return this.options.dev ? ['css-hot-loader'].concat(extraction) : extraction
```
Comment (pi0) [inline lib/builder/webpack/style-loader.js]: Won't this conflict with vue-style-loader?

File: test/basic.dev.test.js [hunk 2]
```diff
@@ -20,7 +20,10 @@ test.serial('Init Nuxt.js', async t => {
     dev: true,
     build: {
       stats: false,
-      profile: true
+      profile: true,
+      extractCSS: {
+        allChunks: true
```
Comment (pi0) [inline test/basic.dev.test.js]: (Not related to this PR) If it is a better approach, what's your idea if we make default behavior of `extractCSS` to extract all chunks?

**Review by pi0 — APPROVED** [review]

File: test/basic.dev.test.js (hunk 2 above)
Comment (clarkdo) [reply test/basic.dev.test.js]: About this, I have discussed with @Atinux several days ago https://github.com/nuxt/nuxt.js/pull/2081#issuecomment-343670353.  

I think bundle all css in aysnc component into a common css may be not good for performance.

File: lib/builder/webpack/style-loader.js (hunk 1 above)
Comment (clarkdo) [reply lib/builder/webpack/style-loader.js]: `css-hot-loader` is only a wrapper to support HMR in `css`, per my test it should work fine in normal cases. Did I miss anything?

Comment (pi0) [reply lib/builder/webpack/style-loader.js]: If you test it locally that's great :) Let's merge and see it in action ;)

Comment (clarkdo) [reply lib/builder/webpack/style-loader.js]: Hope everything goes fine 😄

**Pull request merged**